### ISSUE TITLE
False positive in "no-throw-literal" when `undefined` is redefined

### ISSUE THREAD

**Issue by g-plane** [eslint/eslint#11438]
**Tell us about your environment**

* **ESLint Version:** 5.14.1

**What parser (default, Babel-ESLint, etc.) are you using?**
espree

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```json
{
  "rules": {
    "no-throw-literal: 2
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
var undefined = new Error()
throw undefined
```

**What did you expect to happen?**
No reports.

**What actually happened? Please include the actual, raw output from ESLint.**
Linting-error reported.
Repro: https://eslint.org/demo/#eyJ0ZXh0IjoidmFyIHVuZGVmaW5lZCA9IG5ldyBFcnJvcigpXG50aHJvdyB1bmRlZmluZWQiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjUsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsibm8tdGhyb3ctbGl0ZXJhbCI6Mn0sImVudiI6e319fQ==

**Are you willing to submit a pull request to fix this bug?**
Yes

**Comment by aladdin-add** [comment]
I suggest not report `throw undefined` at all, since `undefined` is not a "Literal".

**Comment by mysticatea** [comment]
Thank you for the report.

However, it looks correct behavior because we cannot redeclare/reassign `undefined` global variable since ES5.

**Comment by g-plane** [comment]
Maybe we can check the `ecmaVersion`?

**Comment by aladdin-add** [comment]
> However, it looks correct behavior because we cannot redeclare/reassign `undefined` global variable since ES5.

it's true, but it can be shadowed:
```js
{var undefined = "foo";}
```

**Comment by mysticatea** [comment]
I recommend to not depend on parser options because every parser has different options. Some parsers such as `babel-eslint` and `@typescript-eslint/parser` doesn't have `ecmaVersion` option, I think.

Yes, we can shadow it.

**Comment by aladdin-add** [comment]
Some background:

the issue comes from https://github.com/aladdin-add/eslint-plugin-autofix/pull/37#discussion_r259670157. I think we just no auto fix this case.

**Comment by platinumazure** [comment]
espree does still support ES3-- should we fix this?

**Comment by aladdin-add** [comment]
+ redeclare/reassign only happens in pre-es5.
+ shadowing can happen in all envrionments, including es5 and pre-es5.

Technically, this is a bug, but I don't have a strong opinion whether to fix it or not. (as far as I can tell,  very few devs using assign `undefined` to some different values.)

should we reopen the issue to see other members' thoughts? 😄

**Comment by platinumazure** [comment]
Apparently nobody else thought this was an important issue. :laugh:
